util/ranger/ranger_test.go

```diff
@@ -2447,3 +2450,236 @@ func TestRangeFallbackForBuildColumnRange(t *testing.T) {
+			filterConds: "[]",
+			resultStr:   "[[\"a\",\"b\"]]",
+		},
+		{
+			indexPos:    0,
+			exprStr:     "a > 'ab'",
+			accessConds: "[gt(test.t.a, ab)]",
+			filterConds: "[gt(test.t.a, ab)]",
+			resultStr:   "[[\"ab\",+inf]]",
+		},
+		{
+			indexPos:    0,
+			exprStr:     "a < 'ab'",
+			accessConds: "[lt(test.t.a, ab)]",
+			filterConds: "[lt(test.t.a, ab)]",
+			resultStr:   "[[-inf,\"ab\")]",
+		},
+		{
+			indexPos:    0,
+			exprStr:     "a >= 'abcd'",
+			accessConds: "[ge(test.t.a, abcd)]",
+			filterConds: "[ge(test.t.a, abcd)]",
+			resultStr:   "[[\"ab\",+inf]]",
+		},
+		{
+			indexPos:    0,
+			exprStr:     "a in ('a', 'b', 'c')",
+			accessConds: "[in(test.t.a, a, b, c)]",
+			filterConds: "[]",
+			resultStr:   "[[\"a\",\"a\"] [\"b\",\"b\"] [\"c\",\"c\"]]",
+		},
+		{
+			indexPos:    0,
+			exprStr:     "a in ('a', 'bbb', 'c')",
+			accessConds: "[in(test.t.a, a, bbb, c)]",
+			filterConds: "[in(test.t.a, a, bbb, c)]",
+			resultStr:   "[[\"a\",\"a\"] [\"bb\",\"bb\"] [\"c\",\"c\"]]",
+		},
+		{
+			indexPos:    0,
+			exprStr:     "a != 'a'",
+			accessConds: "[ne(test.t.a, a)]",
+			filterConds: "[]",
+			resultStr:   "[[-inf,\"a\") (\"a\",+inf]]",
+		},
+		{
+			indexPos:    0,
+			exprStr:     "a != 'aaa'",
+			accessConds: "[]",
+			filterConds: "[ne(test.t.a, aaa)]",
+			resultStr:   "[[NULL,+inf]]",
+		},
+		{
+			indexPos:    0,
+			exprStr:     "a is null",
+			accessConds: "[isnull(test.t.a)]",
+			filterConds: "[]",
+			resultStr:   "[[NULL,NULL]]",
+		},
+		{
+			indexPos:    0,
+			exprStr:     "a is not null",
+			accessConds: "[not(isnull(test.t.a))]",
+			filterConds: "[]",
+			resultStr:   "[[-inf,+inf]]",
+		},
+		{
+			indexPos:    1,
+			exprStr:     "a = 'a' and b < 'b'",
+			accessConds: "[eq(test.t.a, a) lt(test.t.b, b)]",
+			filterConds: "[]",
+			resultStr:   "[[\"a\" -inf,\"a\" \"b\")]",
+		},
+		{
+			indexPos:    1,
+			exprStr:     "a = 'aa' and b < 'bb'",
+			accessConds: "[eq(test.t.a, aa) lt(test.t.b, bb)]",
+			filterConds: "[eq(test.t.a, aa) lt(test.t.b, bb)]",
+			resultStr:   "[[\"aa\" -inf,\"aa\" \"bb\")]",
+		},
+		{
+			indexPos:    1,
+			exprStr:     "a = 'a' and b > 'bb'",
+			accessConds: "[eq(test.t.a, a) gt(test.t.b, bb)]",
+			filterConds: "[gt(test.t.b, bb)]",
+			resultStr:   "[[\"a\" \"bb\",\"a\" +inf]]",
+		},
+		{
+			indexPos:    1,
+			exprStr:     "a = 'aa' and b > 'b'",
+			accessConds: "[eq(test.t.a, aa) gt(test.t.b, b)]",
+			filterConds: "[eq(test.t.a, aa)]",
+			resultStr:   "[(\"aa\" \"b\",\"aa\" +inf]]",
+		},
+		//{
+		//	indexPos:    1,
+		//	exprStr:     "a = 'a' and b is null",
+		//	accessConds: "[eq(test.t.a, a) isnull(test.t.b)]",
+		//	filterConds: "[]",
+		//	resultStr:   "[[\"a\" NULL,\"a\" NULL]]",
+		//},
+		//{
+		//	indexPos:    1,
+		//	exprStr:     "a = 'a' and c is not null",
+		//	accessConds: "[eq(test.t.a, a) not(isnull(test.t.c))]",
+		//	filterConds: "[]",
+		//	resultStr:   "[[\"a\" -inf,\"a\" +inf]]",
+		//},
```

Comment:
Why is commented out?

Reply:
The test seems broken and I'm fixing it.

Reply:
Let me update it.

Reply:
Done.